experimental/python/codegen/codegen/generated_dataclass.py

```diff
@@ -0,0 +1,542 @@
+    elif type.name == "dict":
+        [key_param, value_param] = type.parameters
+
+        assert key_param.name == "str"
+
+        return variable_or_dict_type(value_param)
+    else:
+        name = "VariableOr" if is_required else "VariableOrOptional"
+
+        return GeneratedType(
+            name=name,
+            package="databricks.bundles.core",
+            parameters=[type],
+        )
+
+
+def variable_or_list_type(element_type: GeneratedType) -> GeneratedType:
+    return GeneratedType(
+        name="VariableOrList",
+        package="databricks.bundles.core",
+        parameters=[element_type],
+    )
+
+
+def variable_or_dict_type(element_type: GeneratedType) -> GeneratedType:
+    return GeneratedType(
+        name="VariableOrDict",
+        package="databricks.bundles.core",
+        parameters=[element_type],
+    )
+
+
+def generate_type(ref: str, is_param: bool) -> GeneratedType:
+    if ref.startswith("#/$defs/slice/"):
+        element_ref = ref.replace("#/$defs/slice/", "#/$defs/")
+        element_type = generate_type(
+            ref=element_ref,
+            is_param=is_param,
+        )
+
+        return GeneratedType(
+            name="list",
+            package=None,
+            parameters=[element_type],
+        )
+
+    if ref == "#/$defs/map/string":
+        return dict_type()
+
+    class_name = packages.get_class_name(ref)
+    package = packages.get_package(ref)
+
+    if is_param and package:
+        class_name += "Param"
+
+    return GeneratedType(
+        name=class_name,
+        package=package,
+        parameters=[],
+    )
+
+
+def resource_type() -> GeneratedType:
+    return GeneratedType(
+        name="Resource",
+        package="databricks.bundles.core",
+        parameters=[],
+    )
+
+
+def generate_dataclass(schema_name: str, schema: Schema) -> GeneratedDataclass:
+    print(f"Generating dataclass for {schema_name}")
+
+    fields = list[GeneratedField]()
+    class_name = packages.get_class_name(schema_name)
+
+    for name, prop in schema.properties.items():
+        is_required = name in schema.required
+        field = generate_field(name, prop, is_required=is_required)
+
+        fields.append(field)
+
+    extends = []
+    package = packages.get_package(schema_name)
+
+    assert package
+
+    if is_resource(schema_name):
+        extends.append(resource_type())
+
+    return GeneratedDataclass(
+        class_name=class_name,
+        package=package,
+        description=schema.description,
+        fields=fields,
+        extends=extends,
+        one_ofs=[],
+    )
+
+
+def _get_type_code(generated: GeneratedType, quote: bool = True) -> str:
+    if generated.parameters:
+        parameters = ", ".join(
+            map(lambda x: _get_type_code(x, quote), generated.parameters)
+        )
+
+        return f"{generated.name}[{parameters}]"
+    else:
+        if generated.name[0].islower():
+            return generated.name
```

Comment:
I suspect `str`, `list`, etc. Can you confirm in a comment?

Reply:
Actually it isn't needed anymore, it avoids quoting builtin types, but now we have very little quoting so it's never a problem